File: 11-05-2024/Juggler Sequence.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function Template for C++

class Solution {
  public:
    vector<long long> jugglerSequence(long long n) {
        // Your Code Start
 vector<long long> sequence;
        sequence.push_back(n); 
        
        while (n > 1) {
            if (n % 2 == 1) {
                n = static_cast<long long>(sqrt(n) * n);
            } else {
                n = static_cast<long long>(sqrt(n)); 
            }
            sequence.push_back(n); 
        }
        
        return sequence;
      ///Your Code End 
    }
};


//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        long long n;
        cin >> n;

        Solution ob;
        vector<long long> ans = ob.jugglerSequence(n);
        for (long long u : ans)
            cout << u << " ";
        cout << endl;
    }
    return 0;
}
// } Driver Code Ends

File: 11-05-2024/Juggler Sequence.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(in.readLine());
        while (t-- > 0) {
            long n = Long.parseLong(in.readLine());

            Solution ob = new Solution();
            List<Long> ans = new ArrayList<>();
            StringBuilder out = new StringBuilder();
            ans = ob.jugglerSequence(n);
            for (int i = 0; i < ans.size(); i++) out.append(ans.get(i) + " ");
            System.out.println(out);
        }
    }
}
// } Driver Code Ends




// User function Template for Java

class Solution {
    static List<Long> jugglerSequence(long n) {
 //Your Code Start
       List<Long> sequence = new ArrayList<>();
        sequence.add(n); 
        
        while (1<n) {
            if (n % 2 == 1) {
                 n = (long) (Math.sqrt(n) * n);
            } else {
                n = (long) Math.sqrt(n); 
            }
            sequence.add(n); 
        }
        
        return sequence;
      //Your Code End
    }
}
